src/applications/model/crazytaxi-streaming-server.h

```diff
@@ -0,0 +1,94 @@
+/* -*- Mode:C++; c-file-style:"gnu"; indent-tabs-mode:nil; -*- */
+/*
+ * Copyright (c) 2020, University of Padova, Department of Information
+ * Engineering, SIGNET Lab.
+ *
+ * This program is free software; you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License version 2 as
+ * published by the Free Software Foundation;
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program; if not, write to the Free Software
+ * Foundation, Inc., 59 Temple Place, Suite 330, Boston, MA  02111-1307  USA
+ *
+ * Authors: Salman Mohebi <s.mohebi22@gmail.com>
+ *
+ */
+
+#ifndef CRAZY_TAXI_STREAMING_SERVER_H
+#define CRAZY_TAXI_STREAMING_SERVER_H
+
+#include "ns3/application.h"
+#include "ns3/event-id.h"
+#include "ns3/ptr.h"
+#include "ns3/ipv4-address.h"
+#include "ns3/random-variable-stream.h"
+#include "ns3/gaming-traffic-model.h"
+#include "ns3/gaming-streaming-server.h"
+#include "ns3/callback.h"
+#include <functional>
+
+namespace ns3 {
+
+  class Socket;
+  class Packet;
+
+  /**
+   * Implement the gaming traffic streams for the Crazy Taxi based on following paper:
+   * Manzano, Marc, et al. "Dissecting the protocol and network traffic of the OnLive
+   * cloud gaming platform." Multimedia systems 20.5 (2014): 451-470.
+   */
+  class CrazyTaxiStreamingServer : public GamingStreamingServer
+  {
+  public:
+    /**
+     * \brief Get the type ID.
+     * \return the object TypeId
+     */
+    static TypeId GetTypeId (void);
+
+    CrazyTaxiStreamingServer ();
+    CrazyTaxiStreamingServer (Ipv4Address ip, uint16_t port);
+
+    /**
+     * \brief Add parameters of new stream
+     * \param pkt A random number generator for packet size
+     * \param iat A random number generator for inter-arrival time
+     */
+    void AddNewTraffic (Ptr<RandomVariableStream> pkt, Ptr<RandomVariableStream> iat);
+
+  private:
+    /**
+    *  Initialize the parameters of different streams
+    */
+    void InitializeStreams ();
+
+    /**
+     * \brief Send a packet
+     */
+    void Send (void);
+    /**
+     * \brief Send a packet of a given size
+     * \param size the packet size
+     */
+
+    uint32_t m_sentSeq;
+    uint32_t m_totSentPackets; //!< Counter for sent pack  ets
+    Ptr<Socket> m_socket; //!< Socket
+    Address m_peerAddress; //!< Remote peer address
+    uint16_t m_peerPort; //!< Remote peer port
+    EventId m_sendEvent; //!< Event to send the next packet
+
+    std::vector <Ptr<RandomVariableStream> > m_pksList;  //!< A list of random number generator for packet size
+    std::vector <Ptr<RandomVariableStream> > m_iatList; //!< A list of random number generator for packets inter-arrival time
```

Comment:
most of them are in common with the base class. as written before, think about making them protected instead